### DISCUSSION TITLE
[Proposal] Add Span/ReadOnlySpan CopyTo with an start index position

### DISCUSSION THREAD

**Discussion by WalterDias** [dotnet/csharplang#6281] [Language Ideas] [answered]
### Summary
Add a CopyTo that has an additional parameter to specify a start location;

### Motivation
Often when we are dealing with formated data we need to remove some unanted parts to work(compute/store) with only the "usable" part of it.

### Code Suggetion
```csharp
ReadOnlySpan<char> documentString = "AB123.45C";
Span<char> onlyNumber = stackalloc char[5];

documentString.Slice(2,3).CopyTo(onlyNumber , 0);
documentString.Slice(6,2).CopyTo(onlyNumber , 3);

```
So only number will be present at onlyNumber variable

This is my first suggestion! Thx

**Comment by FaustVX** [comment, marked as answer]
You could just use the `Slice` on the `onlyNumber`
```c#
ReadOnlySpan<char> documentString = "AB123.45C";
Span<char> onlyNumber = stackalloc char[5];

documentString.Slice(2,3).CopyTo(onlyNumber);
documentString.Slice(6,2).CopyTo(onlyNumber[3..]);
```
[sharplab.io](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgZgAIMiBhIgbzyNpPXSprpYCUBTAQzgHkoAbAJ4BlAA6coAHgDGAC05gAfETgB7aRAC27KMGHAwASygBzIgF4iAIgCCAIUzoCAOgAsAVjJWA3Mxa0xCRl5JSJVAUEAOS0AI3YwCyIAZ2BOaQBrTn5+dSI5BQBtdwBdX1w/fzUNbV19I1NnYX5DaXYACnQYAgBKZzJVUUEAFVU28KFozTiwbrL/OiqtHT0DYxNG5ta2gDYYdF7+wZGxiMnpjZb2ntmKlnGo2PjnAEkoJNF2aWA2m/LcFgAvngAUA===)
